src/delegation_backend/app_config.go

```diff
@@ -36,7 +20,11 @@ func LoadEnv(log logging.EventLogger) AppConfig {
 		decoder := json.NewDecoder(file)
 		err = decoder.Decode(&config)
 		if err != nil {
-			log.Fatalf("Error loading config file: %s", err)
+			log.Fatalf("Error decoding config file: %s", err)
+		}
+		if config.Aws != nil {
```

Comment:
I wanted to say that this should go in the `else` of `if configFile != ""` but I see that you also configure it there. I am a bit confused. What is the reason for this being here?

Reply:
Refactored.

Reply:
I don't see this being changed. If this stays like this the if on line 58 would never be true

Reply:
Oh, sorry! It is basically what was happening in `loadAwsCredentials` function. That previous function was loading `aws_creds.json` file and was setting those env variables using data from the file.

```
	os.Setenv("AWS_ACCESS_KEY_ID", credentials.AccessKeyId)
	os.Setenv("AWS_SECRET_ACCESS_KEY", credentials.SecretAccessKey)
```

Now, since we load config from `app_config.json` we set those env variables here. We are setting them only if we have aws config section, hence: `if config.Aws != nil {`

In `else` we have to don't have configFile so we need to have those env variables already set. So there we do `os.Getenv("AWS_ACCESS_KEY_ID")` etc.

Reply:
Yeah, makes sense, got confused a bit.